src/main/scala/com/fortysevendeg/scala/android/modules/utils/validation.scala

```diff
@@ -0,0 +1,40 @@
+package com.fortysevendeg.scala.android.modules.utils
+
+import android.widget.EditText
+import com.fortysevendeg.scala.android.R
+import macroid.ContextWrapper
+import com.fortysevendeg.macroid.extras.ResourcesExtras._
+
+import scalaz._
+import scalaz.std.list._
+import scalaz.syntax.traverse._
+
+object validation {
+
+  type Validator[T] = (T) => Validation[ValidationMessage, T]
+
+  case class ValidationMessage(
+      widget: Option[EditText] = None,
+      message: String,
+      details: String = "")
+
+  case class ValidationException(
+      errorHead: ValidationMessage,
+      errorTail: Seq[ValidationMessage] = Seq.empty)
+      extends RuntimeException(s"${errorHead.message} - ${errorHead.details}")
+
+  def WithValidation[T, ValT <: T, A, S]
+  (validators: ((ValT) => Validation[ValidationMessage, T])*)
+      (happyPath: => S)
+      (implicit input: ValT, m: Manifest[T], contextWrapper: ContextWrapper): S = {
+
+    validators.toList.map(p => p(input).toValidationNel).sequenceU match {
+      case Failure(NonEmptyList(head: ValidationMessage)) => throw ValidationException(head)
```

Comment:
I don't see the point of pattern matching on types like this if we are already doing a constructor match. Expected something like `Failure(NonEmptyList("error1", "error2")) match { case Failure(NonEmptyList(err, _*)) => err  }`